plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java

```diff
@@ -0,0 +1,118 @@
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Closer;
+import org.testcontainers.containers.Network;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.Network.newNetwork;
+
+public class HiveMinioDataLake
+        implements Closeable, AutoCloseable
+{
+    public static final String ACCESS_KEY = "accesskey";
+    public static final String SECRET_KEY = "secretkey";
+
+    private final String bucketName;
+    private final Minio minio;
+    private final HiveHadoop hiveHadoop;
+
+    private final Closer closer = Closer.create();
+
+    public HiveMinioDataLake(String bucketName, Map<String, String> hiveHadoopFilesToMount)
+    {
+        this(bucketName, hiveHadoopFilesToMount, HiveHadoop.DEFAULT_IMAGE);
+    }
+
+    public HiveMinioDataLake(String bucketName, Map<String, String> hiveHadoopFilesToMount, String hiveHadoopImage)
+    {
+        this.bucketName = requireNonNull(bucketName, "bucketName is null");
+
+        Network network = newNetwork();
+        this.closer.register(network::close);
+
+        this.minio = closer.register(
+                Minio.builder()
+                        .withNetwork(network)
+                        .withEnvVars(ImmutableMap.<String, String>builder()
+                                .put("MINIO_ACCESS_KEY", ACCESS_KEY)
+                                .put("MINIO_SECRET_KEY", SECRET_KEY)
+                                .build())
+                        .build());
+
+        this.hiveHadoop = closer.register(
+                HiveHadoop.builder()
+                        .withFilesToMount(ImmutableMap.<String, String>builder()
+                                .put("hive_s3_insert_overwrite/hive-core-site.xml", "/etc/hadoop/conf/core-site.xml")
+                                .putAll(hiveHadoopFilesToMount)
+                                .build())
+                        .withImage(hiveHadoopImage)
+                        .withNetwork(network)
+                        .build());
+    }
+
+    public void start()
+    {
+        this.minio.start();
+        this.hiveHadoop.start();
+        AmazonS3 s3Client = AmazonS3ClientBuilder
+                .standard()
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(
+                        "http://localhost:" + minio.getMinioApiEndpoint().getPort(),
+                        "us-east-1"))
+                .withPathStyleAccessEnabled(true)
+                .withCredentials(new AWSStaticCredentialsProvider(
+                        new BasicAWSCredentials(ACCESS_KEY, SECRET_KEY)))
+                .build();
+        s3Client.createBucket(this.bucketName);
+    }
+
+    public void stop()
+    {
+        try {
```

Comment:
nit: should you keep a "alreadyStopped" flag and make this no-op if called more than once?